### PR TITLE
[Docs] Add docs for distinct metric and unified_transformer.

### PR DESCRIPTION
**Pull request by xiemoyuan**
### PR types
Others

### PR changes
APIs

### Description
Add docs for distinct metric and unified_transformer.

### CODE REVIEW

File: paddlenlp/metrics/distinct.py [hunk 1]
```diff
@@ -25,23 +25,26 @@ class Distinct(paddle.metric.Metric):
     the value of n-grams, the higher the diversity of the text. See detail at
```
Comment (yingyibiao) [inline paddlenlp/metrics/distinct.py]: 第一个单词Distinct是不是加点特殊符号来强调一下

Comment (yingyibiao) [reply paddlenlp/metrics/distinct.py]: The larger the value of n-grams 表达不太准确

Comment (yingyibiao) [reply paddlenlp/metrics/distinct.py]: detail -> details

File: paddlenlp/metrics/distinct.py [hunk 2]
```diff
@@ -25,23 +25,26 @@ class Distinct(paddle.metric.Metric):
     the value of n-grams, the higher the diversity of the text. See detail at 
     https://arxiv.org/abs/1510.03055
 
-    `Distinct` could be used as `paddle.metric.Metric` class, or an ordinary 
-    class. When `Distinct` is used as `paddle.metric.Metric` class. A function is
-    needed that transforms the network output to string list. It should be noted 
-    that the `Distinct` here is different from the `Distinct` calculated in 
-    prediction, and it is only for observation during training and evaluation.
+    :class:`Distinct` could be used as :class:`paddle.metric.Metric` class, or
```
Comment (yingyibiao) [inline paddlenlp/metrics/distinct.py]: be used as -> be used as a

File: paddlenlp/metrics/distinct.py [hunk 3]
```diff
@@ -25,23 +25,26 @@ class Distinct(paddle.metric.Metric):
...
     https://arxiv.org/abs/1510.03055
 
-    `Distinct` could be used as `paddle.metric.Metric` class, or an ordinary 
-    class. When `Distinct` is used as `paddle.metric.Metric` class. A function is
-    needed that transforms the network output to string list. It should be noted 
-    that the `Distinct` here is different from the `Distinct` calculated in 
-    prediction, and it is only for observation during training and evaluation.
+    :class:`Distinct` could be used as :class:`paddle.metric.Metric` class, or 
+    an ordinary class. When :class:`Distinct` is used as 
+    :class:`paddle.metric.Metric` class, a function is needed that transforms
```
Comment (yingyibiao) [inline paddlenlp/metrics/distinct.py]: is used as -> is used as a

Comment (yingyibiao) [reply paddlenlp/metrics/distinct.py]: that transforms -> to transform

File: paddlenlp/metrics/distinct.py [hunk 4]
```diff
@@ -25,23 +25,26 @@ class Distinct(paddle.metric.Metric):
...
 
-    `Distinct` could be used as `paddle.metric.Metric` class, or an ordinary 
-    class. When `Distinct` is used as `paddle.metric.Metric` class. A function is
-    needed that transforms the network output to string list. It should be noted 
-    that the `Distinct` here is different from the `Distinct` calculated in 
-    prediction, and it is only for observation during training and evaluation.
+    :class:`Distinct` could be used as :class:`paddle.metric.Metric` class, or 
+    an ordinary class. When :class:`Distinct` is used as 
+    :class:`paddle.metric.Metric` class, a function is needed that transforms 
+    the network output to string list. It should be noted that the
```
Comment (yingyibiao) [inline paddlenlp/metrics/distinct.py]: to string list ->  to a string list

File: paddlenlp/metrics/distinct.py [hunk 5]
```diff
@@ -25,23 +25,26 @@ class Distinct(paddle.metric.Metric):
...
-    class. When `Distinct` is used as `paddle.metric.Metric` class. A function is
-    needed that transforms the network output to string list. It should be noted 
-    that the `Distinct` here is different from the `Distinct` calculated in 
-    prediction, and it is only for observation during training and evaluation.
+    :class:`Distinct` could be used as :class:`paddle.metric.Metric` class, or 
+    an ordinary class. When :class:`Distinct` is used as 
+    :class:`paddle.metric.Metric` class, a function is needed that transforms 
+    the network output to string list. It should be noted that the 
+    :class:`Distinct` here is different from the :class:`Distinct` calculated 
+    in prediction, and it is only for observation during training and evaluation.
```
Comment (yingyibiao) [inline paddlenlp/metrics/distinct.py:32]: 这里写成Note:块比较好一些。
btw，这句话不太好理解。

File: paddlenlp/metrics/distinct.py [hunk 6]
```diff
@@ -25,23 +25,26 @@ class Distinct(paddle.metric.Metric):
...
 
     Args:
         trans_func (callable, optional): `trans_func` transforms the network
-            output to string list. Default None. When `Distinct` is used as 
-            `paddle.metric.Metric` class, `trans_func` must be provided. Please
-            note that the input of `trans_func` is numpy array.
-        n_size (int, optional): Number of gram for `Distinct` metric. Default: 2.
-        name (str, optional): Name of `paddle.metric.Metric` instance.
+            output to string list. Default None. When :class:`Distinct` is used
```
Comment (yingyibiao) [inline paddlenlp/metrics/distinct.py]: to string list -> to a string list

Comment (yingyibiao) [reply paddlenlp/metrics/distinct.py]: Default:  None

Comment (yingyibiao) [reply paddlenlp/metrics/distinct.py]: 这里也加一个 Note:块比较清晰一点

File: paddlenlp/metrics/distinct.py [hunk 7]
```diff
@@ -126,7 +130,9 @@ def accumulate(self):
         return distinct
 
     def score(self):
+        """The function is the same as :meth:`accumulate` method."""
         return self.accumulate()
 
     def name(self):
+        """Returns the metric name."""
         return self._name
```
Comment (yingyibiao) [inline paddlenlp/metrics/distinct.py:160]: 上述方法有输入，输出的话都需要明确写出来。另外注意首句的第三人称单数。

File: paddlenlp/transformers/unified_transformer/modeling.py [hunk 8]
```diff
@@ -152,6 +154,76 @@ def forward(self, input_ids, token_type_ids, position_ids):
...
+            `hidden_size`. Defaults to 3072.
+        hidden_act (str, optional):
+            The activation function in the feedforward network. Defaults to 
+            "gelu".
+        hidden_dropout_prob(float, optional): 
+            The dropout probability used in pre-process and post-precess of MHA 
+            and FFN sub-layer. Default to 0.1.
+        attention_probs_dropout_prob (float, optional): 
+            The dropout probability used in MHA to drop some attention target. 
+            Default to 0.1.
```
Comment (yingyibiao) [inline paddlenlp/transformers/unified_transformer/modeling.py]: Default to -> Defaults to 下同

File: paddlenlp/transformers/unified_transformer/modeling.py [hunk 9]
```diff
@@ -267,6 +418,52 @@ def forward(self,
...
+        Returns:
+            Tensor|tuple[List]: If `use_cache` is True, it is a tensor 
+            representing the output of :class:`UnifiedTransformerLMHeadModel`, 
+            with shape [batch_size, sequence_length, vocab_size]. The data type 
+            is float32 or float64. Otherwise, it is a tuple, besides the output 
+            of :class:`UnifiedTransformerLMHeadModel`, the tuple also includes 
+            the new cache which is same as input `cache` but `incremental_cache` 
+            in it has an incremental length. 
+            See :meth:`paddle.nn.MultiHeadAttention.gen_cache` method and 
+            :meth:`MultiHeadAttention.forward` method for more details.
```
Comment (yingyibiao) [inline paddlenlp/transformers/unified_transformer/modeling.py]: :meth:`MultiHeadAttention.forward` -> :meth:`paddle.nn.MultiHeadAttention.forward`

File: paddlenlp/transformers/unified_transformer/modeling.py [hunk 10]
```diff
@@ -203,6 +275,76 @@ def forward(self,
...
+        Returns:
+            Tensor|tuple[List]: If `use_cache` is True, it is a tensor 
+            representing the output of :class:`UnifiedTransformerModel`, with 
+            shape [batch_size, sequence_length, hidden_size]. The data type is 
+            float32 or float64. Otherwise, it is a tuple, besides the output of 
+            :class:`UnifiedTransformerModel`, the tuple also includes the new 
+            cache which is same as input `cache` but `incremental_cache` in it 
+            has an incremental length. 
+            See :meth:`paddle.nn.MultiHeadAttention.gen_cache` method and 
+            :meth:`MultiHeadAttention.forward` method for more details.
```
Comment (yingyibiao) [inline paddlenlp/transformers/unified_transformer/modeling.py]: :meth:`MultiHeadAttention.forward` -> :meth:`paddle.nn.MultiHeadAttention.forward`

File: paddlenlp/transformers/unified_transformer/modeling.py [hunk 11]
```diff
@@ -203,6 +275,76 @@ def forward(self,
...
+            attention_mask (Tensor): 
+                A tensor used in multi-head attention to prevents attention to 
+                some unwanted positions, usually the paddings or the subsequent 
+                positions. It is a tensor with shape broadcasted to 
+                [batch_size, n_head, sequence_length, sequence_length]. When 
+                the data type is bool, the unwanted positions have `False` 
+                values and the others have `True` values. When the data type is 
+                int, the unwanted positions have 0 values and the others have 1 
+                values. When the data type is float, the unwanted positions have 
+                `-INF` values and the others have 0 values.
```
Comment (yingyibiao) [inline paddlenlp/transformers/unified_transformer/modeling.py]: 这块太长了，建议分条目写，清晰一点。

File: paddlenlp/transformers/unified_transformer/tokenizer.py [hunk 12]
```diff
@@ -198,19 +246,32 @@ def _tokenize(self, text, is_split_into_words=True):
...
         Args:
-            text (str): The text to be tokenized.
-            is_split_into_words(bool, optinal): Whether or not the input `text` 
-                has been pretokenized. Default True.
+            text (str): 
+                The text to be tokenized.
+            is_split_into_words (bool, optinal): 
+                Whether or not the input `text` has been pretokenized. If False, 
+                the input `text` will be pretokenized by `jieba` firstly. 
+                Default True.
```
Comment (yingyibiao) [inline paddlenlp/transformers/unified_transformer/tokenizer.py]: Default True. -> Defaults to True. 下同

File: paddlenlp/transformers/unified_transformer/tokenizer.py [hunk 13]
```diff
@@ -514,7 +519,60 @@ def dialogue_encode(self,
             is_split_into_words(bool, optinal): Whether or not the input text 
                 (`history`, `response` and `knowledge`) has been pretokenized. 
                 Default True.
+
+        Returns: 
+            Dict: A dictionary containing the encoded sequence and other
```
Comment (yingyibiao) [inline paddlenlp/transformers/unified_transformer/tokenizer.py]: 这里的Dict显示上会链接到paddlenlp.data

Comment (yingyibiao) [reply paddlenlp/transformers/unified_transformer/tokenizer.py]: [CLS]是str，需要加上""

Comment (yingyibiao) [reply paddlenlp/transformers/unified_transformer/tokenizer.py]: 参数默认值： Default -> Defaults to

Comment (xiemoyuan) [reply paddlenlp/transformers/unified_transformer/tokenizer.py]: Done. Thanks.

File: paddlenlp/transformers/unified_transformer/tokenizer.py (hunk 12 above)
Comment (xiemoyuan) [reply paddlenlp/transformers/unified_transformer/tokenizer.py]: Done. Thanks.

File: paddlenlp/transformers/unified_transformer/modeling.py (hunk 11 above)
Comment (xiemoyuan) [reply paddlenlp/transformers/unified_transformer/modeling.py]: Done. Thanks.

File: paddlenlp/transformers/unified_transformer/modeling.py (hunk 10 above)
Comment (xiemoyuan) [reply paddlenlp/transformers/unified_transformer/modeling.py]: Done. Thanks.

File: paddlenlp/transformers/unified_transformer/modeling.py (hunk 9 above)
Comment (xiemoyuan) [reply paddlenlp/transformers/unified_transformer/modeling.py]: Done. Thanks.

File: paddlenlp/transformers/unified_transformer/modeling.py (hunk 8 above)
Comment (xiemoyuan) [reply paddlenlp/transformers/unified_transformer/modeling.py]: Done. Thanks.

File: paddlenlp/metrics/distinct.py (hunk 7 above)
Comment (xiemoyuan) [reply paddlenlp/metrics/distinct.py:160]: Done. Thanks.

File: paddlenlp/metrics/distinct.py (hunk 6 above)
Comment (xiemoyuan) [reply paddlenlp/metrics/distinct.py]: Done. Thanks.

File: paddlenlp/metrics/distinct.py (hunk 5 above)
Comment (xiemoyuan) [reply paddlenlp/metrics/distinct.py:32]: Done. Thanks.

File: paddlenlp/metrics/distinct.py (hunk 4 above)
Comment (xiemoyuan) [reply paddlenlp/metrics/distinct.py]: Done. Thanks.

File: paddlenlp/metrics/distinct.py (hunk 3 above)
Comment (xiemoyuan) [reply paddlenlp/metrics/distinct.py]: Done. Thanks.

File: paddlenlp/metrics/distinct.py (hunk 2 above)
Comment (xiemoyuan) [reply paddlenlp/metrics/distinct.py]: Done. Thanks.

File: paddlenlp/metrics/distinct.py (hunk 1 above)
Comment (xiemoyuan) [reply paddlenlp/metrics/distinct.py]: Done. Thanks.

File: paddlenlp/metrics/distinct.py [hunk 14]
```diff
@@ -20,28 +20,32 @@
...
+    details at https://arxiv.org/abs/1510.03055.
 
-    `Distinct` could be used as `paddle.metric.Metric` class, or an ordinary 
-    class. When `Distinct` is used as `paddle.metric.Metric` class. A function is
-    needed that transforms the network output to string list. It should be noted 
-    that the `Distinct` here is different from the `Distinct` calculated in 
-    prediction, and it is only for observation during training and evaluation.
+    :class:`Distinct` could be used as a :class:`paddle.metric.Metric` class, 
+    or an ordinary class. When :class:`Distinct` is used as a 
+    :class:`paddle.metric.Metric` class, a function is needed to transforms
```
Comment (yingyibiao) [inline paddlenlp/metrics/distinct.py]: transforms -> transform

**Review by yingyibiao — COMMENTED** [review]
typo

File: paddlenlp/metrics/distinct.py (hunk 14 above)
Comment (xiemoyuan) [reply paddlenlp/metrics/distinct.py]: Done. Thanks.

**Review by yingyibiao — APPROVED** [review]
LGTM
